pennylane/pauli/pauli_arithmetic.py

```diff
@@ -202,6 +205,40 @@ def __mul__(self, other):
 
         return PauliWord(result), coeff
 
+    def __mul__(self, other):
+        """Multiply a PauliWord by a scalar
+
+        Args:
+            other (Scalar): The scalar to multiply the PauliWord with
+
+        Returns:
+            PauliSentence
+        """
+        if isinstance(other, PauliWord):
+            # this is legacy support and will be removed after a deprecation cycle
+            return self @ other
+
+        if isinstance(other, TensorLike):
+            if not qml.math.ndim(other) == 0:
+                raise ValueError(
+                    f"Attempting to multiply a PauliWord with an array of dimension {qml.math.ndim(other)}"
+                )
+
+            return PauliSentence({self: other})
+        raise TypeError(
+            f"PauliWord can only be multiplied by numerical data. Attempting to multiply by {other} of type {type(other)}"
+        )
+
+    __rmul__ = __mul__
```

Comment:
I'm not sure we should do this quite yet, or add a custom implementation to `__rmul__`, since `pw1 * pw2 != pw2 * pw1`.

Reply:
good point!

Reply:
Though this is actually fine I think:
When multiplying two PauliWords it is never retreating to the rmul because a mul between two pauli words is defined. The outputs are correct as expected:
```python
pauliX = PauliWord({0:"X"})
pauliY = PauliWord({0:"Y"})

>>> pauliX * pauliY, pauliY * pauliX
((Z(0), 1j), (Z(0), -1j))
```

Same for PauliSentence:
```python
pauliX = PauliSentence({PauliWord({0:"X"}):1})
pauliY = PauliSentence({PauliWord({0:"Y"}):1})

>>> pauliX * pauliY, pauliY * pauliX
(1j * Z(0), -1j * Z(0))
```

For the case where rmul would be invoked, e.g. between pauli word and paulisentence, `pw * ps`, there is no legacy support since this was not part of the pauli arithmetic beforehand, anyway.

Note that the motivation for adding `__rmul__` here is mainly to be able to multiply scalars from the left `1. * pw/ps`

Reply:
I added two tests that explicitly test the non-commutativity for the legacy support

Reply:
good point! thanks.